xt/testapp/t/004_selenium.t

```diff
@@ -0,0 +1,312 @@
+
+The method used to find the "post_login_page_btn" on the IdP login page. The supported
+methods are: class, class_name, css, id, link, link_text, partial_link_text,
+tag_name, name, xpath.
+
+=item * L<post_logout_page_title>
+
+Some IdPs provide a web page that is displayed after logout and this allows the
+script to recoginize the page and move past it to the Net::SAML2 testapp login page.
+
+=item * L<login_bindings>
+
+Array of login bindings that the IdP supports (supported are: post, redirect).
+
+=item * L<logout_bindings>
+
+Array of logout bindings that the IdP supports (supported are: local, post, redirect).
+
+=item * L<issuer_value>
+
+Value of the issuer to find on the user logged in page for Net::SAML2 testapp to
+confirm that the login worked
+
+=back
+
+=cut
+if (! -d "$RealBin/selenium" ) {
+    warn "Created directory $RealBin/selenium\n" if (make_path("$RealBin/selenium"));
+}
+
+if (! -e -f "$RealBin/selenium/chromedriver") {
+    warn "Please ensure that the chromedriver binary is in $RealBin/selenium/\n";
+    exit;
+}
+
+my $driver =  Selenium::Chrome->new(
+                        binary => "$RealBin/selenium/chromedriver",
+                        accept_ssl_certs  => 1,
+                        );
+
+isa_ok($driver, 'Selenium::Chrome');
+
+my $ret = $driver->get('https://netsaml2-testapp.local');
+if ( ! $ret ) {
+    die "Unable to access https://netsaml2-testapp.local";
+}
+
+ok($ret, "Access https://netsaml2-testapp.local");
+
+##############################################
+# FIXME: Get this based on the list in IdPs
+##############################################
+
+ok ( -d -e 'IdPs', "'IdPs' directory exists");
+
+my @idps = load_idps();
+
+##############################################
+# Loop through each of the IdPs being tested
+##############################################
+foreach my $idp (@idps) {
+    subtest "Authenticate and logout at each IdP - $idp" => \&auth_to_idp, $idp;
+}
+
+$driver->quit;
+
+sub auth_to_idp {
+    my $idp = shift;
+
+    note "==============================================\n";
+    note "= Beginning test for IdP: $idp\n";
+    note "==============================================\n";
+    # Load the IdP configuration file
+    my $config_file = catfile( "IdPs/$idp/selenium.yml");
+    if ( ! -e $config_file ) {
+        plan skip_all => "IdPs/$idp/selenium.yml does not exist";
+        next;
+    }
+    ok (defined $config_file, "selenium config file found for $idp");
+
+    my $selenium_config = YAML::LoadFile($config_file);
+    if ( ! $selenium_config ) {
+        die "Undefined selenium_config";
+    }
+    ok ($selenium_config, "Loaded selenium config from $config_file");
+
+    ############################################
+    # Loop through each of the logout bindings
+    ############################################
+    foreach my $logout_binding (@{$selenium_config->{"logout_bindings"}}) {
+        note "Logout: ", $logout_binding, "\n";
```

Comment:
I believe `note` is always showing when running the testsuite and `diag` is only when running with `-v`.